src/julia.h

```diff
@@ -1055,13 +1114,13 @@ STATIC_INLINE jl_value_t *jl_array_ptr_set(
 STATIC_INLINE uint8_t jl_array_uint8_ref(void *a, size_t i) JL_NOTSAFEPOINT
 {
     assert(i < jl_array_len(a));
-    assert(jl_typeis(a, jl_array_uint8_type));
+    assert(jl_typetagis(a, jl_array_uint8_type));
```

Comment:
```suggestion
    assert(jl_typeis(a, jl_array_uint8_type));
```

Reply:
`jl_typetagis` is always faster than `jl_typeis`, which is kept as a legacy alias, if you can guarantee that the type is not sometimes represented as a tag.